tools/libSwiftSyntaxParser/CMakeLists.txt

```diff
@@ -20,6 +20,13 @@ add_llvm_symbol_exports(libSwiftSyntaxParser ${LLVM_EXPORTED_SYMBOL_FILE})
 # Adds -dead_strip option
 add_link_opts(libSwiftSyntaxParser)
 
+if(LIBPARSER_VER)
+  set(LIBPARSER_LINK_FLAGS " -Wl,-compatibility_version -Wl,1")
```

Comment:
2 improvements:
* Add `SWIFT_` prefix to the CMake variable
* Those link flags are only valid for macOS, add handling of the version for other platforms by following what libclang is doing here: https://github.com/apple/swift-clang/blob/38ef19dd1f3ea3d09a8b7e77b29aa95fd6ad7d87/tools/libclang/CMakeLists.txt#L108

Reply:
Good point! Done. 
Do we need  `DEFINE_SYMBOL _CINDEX_LIB_` here? does SwiftSyntax parser have a similar symbol?

Reply:
No you can ignore this, for the parser we just leave it to CMake to automatically define the right macro, which is then checked here: https://github.com/apple/swift/blob/b5299531d4e498002920ebb0ddf853e12212b0e6/include/swift-c/SyntaxParser/SwiftSyntaxParser.h#L58